packages/common/src/Adaptor.js

```diff
@@ -200,36 +200,47 @@ export function lastReferenceValue(path) {
  * Scopes an array of data based on a JSONPath.
```

Comment:
Can we re-write these docs from scratch? I don't think it's a very  clear explanation of what this function is actually doing.